kokiri/src/graphics/opengl/renderer2d.hpp

```diff
@@ -4,18 +4,40 @@
 #include <GL/glext.h>
 #endif
 
+#include <SDL2/SDL.h>
 
-#include "context.hpp"
-
-#include "../window.hpp"
+#include "core/window.hpp"
+#include "core/references.hpp"
 
 namespace Kokiri {
     namespace Graphics {
         namespace OpenGL {
             class Renderer2D {
+            private:
+                // An instance of the OpenGLContext which holds all information of the
+                // SDL context. One thing that should be noted is that this is not
+                // verified to be a good encapsulation at this moment.
+                Core::Shared<SDL_GLContext> m_context;
+
             public:
-                Renderer2D(Window &&window);
-                Renderer2D(Window &&window, Context::ContextVersion version);
+                /// Current available types of context that cold be created.
+                enum class Type {
+                    CORE,
+                    COMPABILITY
+                };
+
+                /// Describe the current version of OpenGL versions that could
+                /// be used to create the context.
+                enum class Version {
```

Comment:
why only these? @rafaelcn

Reply:
Because I don't care about the other ones. It mostly captures all of the major core profile versions so I'm fine with that.